ggml-raw/build.rs

```diff
@@ -10,13 +9,64 @@ fn main() {
 
     let build = builder.files(ggml_src.iter()).include("include");
 
-    // TODO: This is currently hardcoded for (my) linux.
-    build.flag("-mavx2");
-    build.flag("-mavx");
-    build.flag("-mfma");
-    build.flag("-mf16c");
+    // This is a very basic heuristic for applying compile flags.
+    // Feel free to update this to fit your operating system.
+    let target_arch = std::env::var("CARGO_CFG_TARGET_ARCH").unwrap();
+    let target_os = std::env::var("CARGO_CFG_TARGET_OS").unwrap();
+    let is_release = std::env::var("PROFILE").unwrap() == "release";
+
+    let supported_features: HashSet<_> = std::env::var("CARGO_CFG_TARGET_FEATURE")
```

Comment:
If this new version includes the changes from #8, we need to adjust the README to include instructions.

I think it's okay to merge this for the time being, but I would say in the long run:

- We use something similar to llama.cpp's autodetection (shelling out to commands to autodetect CPU features).
- We allow overriding those heuristics via some env variable, where the user specifies exactly what they want. Useful for cross-compilation.

This makes the default simple for the vast majority of use cases, and if well documented, it wouldn't make things much more complex for people attempting cross-compilation.

Reply:
We can use the improved #8, or I can remove #8 from this PR. Either one is fine by me.